pom.xml

```diff
@@ -124,4 +166,83 @@
     </plugins>
   </reporting>
 
+  <profiles>
+    <profile> 
+      <id>release</id>
+      <build>
+        <plugins>
+          <plugin>
+            <!-- Package the source jar. -->
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-source-plugin</artifactId>
+            <version>2.2.1</version>
+            <executions>
+              <execution>
+                <id>attach-sources</id>
+                <goals>
+                  <goal>jar-no-fork</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+          <plugin>
+            <!-- Package the javadoc jar. -->
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-javadoc-plugin</artifactId>
+            <version>2.9.1</version>
```

Comment:
I just realized http://www.javadoc.io/doc/software.amazon.ion/ion-java/1.0.0 exists. I'm actually leaning toward just deleting the whole reporting section, since the javadocs are now generated automatically with each commit. The only benefit of the reporting section is that it can be run anytime without committing changes, but it requires `mvn site` to be run explicitly.

I'm interested to hear thoughts.

(Unrelatedly, I think we should remove the generated javadoc html from ion-docs and point all links to http://www.javadoc.io/doc/software.amazon.ion/ion-java/1.0.0 -- I'll put out a PR for that.)